### PR TITLE
/script/p2sh API method got back, minor code refactoring 

### PR DESCRIPTION
**Pull request by kushti**
In this PR:
* P2SH address support back (script/p2sh)
* a bit of unused code removed

### CODE REVIEW

File: src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala [hunk 1]
```diff
@@ -101,6 +101,138 @@ case class ErgoTransaction(override val inputs: IndexedSeq[Input],
...
+                           outAssets: Map[ByteBuffer, Long],
+                           outAssetsNum: Int,
+                           boxesToSpend: IndexedSeq[ErgoBox],
+                           stateContext: ErgoStateContext): ValidationResult[Long] = {
+    // Cost limit per block
+    val maxCost = stateContext.currentParameters.maxBlockCost
+
+    ErgoBoxAssetExtractor.extractAssets(boxesToSpend) match {
+      case Success((inAssets, inAssetsNum)) =>
+        lazy val newAssetId = ByteArrayWrapper(inputs.head.boxId)
```
Comment (pragmaxim) [inline src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala:208]: Using `ByteBuffer.wrap` here would be consistent with the asset-cost PR of mine

Comment (pragmaxim) [reply src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala:208]: This is a bug, as the `ErgoBoxAssetExtractor.extractAssets` returns `ByteBuffer`, so you need to do `ByteBuffer.wrap` I think

File: src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala [hunk 2]
```diff
@@ -101,6 +101,138 @@ case class ErgoTransaction(override val inputs: IndexedSeq[Input],
...
+                         (implicit verifier: ErgoInterpreter): ValidationResult[Long] = {
+
+    // Cost limit per block
+    val maxCost = stateContext.currentParameters.maxBlockCost
+
+    val input = inputs(inputIndex)
+
+    // Just in case, should always be true if client implementation is correct.
+    if(!box.id.sameElements(input.boxId)) {
+      log.error("Critical client error: box is not inputs(inputIndex)")
```
Comment (greenhat) [inline src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala:134]: I wonder what is the reason to "downgrade" this check from `validEquals` which fails the whole validation to `log.error`?

**Review by greenhat — APPROVED** [review]

File: src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala (hunk 2 above)
Comment (aslesarenko) [reply src/main/scala/org/ergoplatform/modifiers/mempool/ErgoTransaction.scala:135]: I don't think it is safe to change the validation logic. What is the motivation?

**Review by aslesarenko — COMMENTED** [review]
Looks good, except one place.

**Review by ccellado — APPROVED** [review]

**Pull request merged**